### PR TITLE
OCPBUGS-35444: Add promotion stanza to coredns-ocp-dnsnameresolver ci config

### PR DESCRIPTION
**Pull request by arkadeepsen**
This PR adds promotion stanza to coredns-ocp-dnsnameresolver ci config

### CODE REVIEW

File: core-services/prow/02_config/openshift/coredns-ocp-dnsnameresolver/_prowconfig.yaml [hunk 1]
```diff
@@ -36,3 +54,19 @@ tide:
     - needs-rebase
     repos:
     - openshift/coredns-ocp-dnsnameresolver
+  - includedBranches:
+    - release-4.17
```
Comment (alebedev87) [inline core-services/prow/02_config/openshift/coredns-ocp-dnsnameresolver/_prowconfig.yaml:58]: You didn't add the config of `release-4.17` branch but added the lables requirements, why?

Comment (arkadeepsen) [reply core-services/prow/02_config/openshift/coredns-ocp-dnsnameresolver/_prowconfig.yaml:58]: I actually am not sure whether the config for release-4.17 and 4.18 will automatically get added or they are needed to be added manually. I think I can probably remove the label requirements for 4.17 as I have not added the config for it.

Comment (arkadeepsen) [reply core-services/prow/02_config/openshift/coredns-ocp-dnsnameresolver/_prowconfig.yaml]: I have removed the label requirements for release-4.17.

**Comment by arkadeepsen** [comment]
/test pull-ci-openshift-coredns-ocp-dnsnameresolver-main-build
/pj-rehearse pull-ci-openshift-coredns-ocp-dnsnameresolver-main-build-operator
/pj-rehearse pull-ci-openshift-coredns-ocp-dnsnameresolver-main-unit
/pj-rehearse pull-ci-openshift-coredns-ocp-dnsnameresolver-main-unit-operator
/pj-rehearse pull-ci-openshift-coredns-ocp-dnsnameresolver-main-verify
/pj-rehearse pull-ci-openshift-coredns-ocp-dnsnameresolver-main-verify-operator

**Comment by arkadeepsen** [comment]
/pj-rehearse pull-ci-openshift-coredns-ocp-dnsnameresolver-main-build

**Comment by arkadeepsen** [comment]
/pj-rehearse pull-ci-openshift-coredns-ocp-dnsnameresolver-main-build-operator pull-ci-openshift-coredns-ocp-dnsnameresolver-main-unit pull-ci-openshift-coredns-ocp-dnsnameresolver-main-unit-operator pull-ci-openshift-coredns-ocp-dnsnameresolver-main-verify pull-ci-openshift-coredns-ocp-dnsnameresolver-main-verify-operator

**Pull request merged**